File: Classes/UserRecord.cpp
//
//  UserRecord.cpp
//  FlappyBird
//
//  Created by 夏旸 on 16/7/23.
//
//

#include "UserRecord.h"


UserRecord::UserRecord(){}
UserRecord::~UserRecord(){}

UserRecord *UserRecord::sharedUserRecord = nullptr;

UserRecord *UserRecord::getInstance()
{
    if (sharedUserRecord == NULL) {
        sharedUserRecord = new UserRecord();
        if (!sharedUserRecord->init()) {
            delete sharedUserRecord;
            sharedUserRecord = NULL;
            CCLOG("ERROR: Could not init sharedInstance");
        }
    }
    
    return sharedUserRecord;
}

void UserRecord::destroyInstance()
{
    CC_SAFE_DELETE(sharedUserRecord);
}

bool UserRecord::init()
{
    return true;
}
int UserRecord::readIntegerFromUserDefault(const char *key)
{
    return UserDefault::getInstance()->getIntegerForKey(key);
}

void UserRecord::saveIntegerToUserDefault(const char *key, int score)
{
    int oldScore = UserDefault::getInstance()->getIntegerForKey(key);
    if (score > oldScore) {
        UserDefault::getInstance()->setIntegerForKey(key, score);
    }
}

File: src/app/tools/UserRecord.lua

local _userRecord = nil

local UserRecord = class("UserRecord", cc.Object)

function UserRecord:getInstance()
    if _userRecord == nil then
        _userRecord = UserRecord:create()
    end
    return _userRecord
end

function UserRecord:destroyInstance()
	_userRecord = nil
end

function UserRecord:saveIntegerToUserDefault(key, score)
	local oldScore = cc.UserDefault:getInstance():getIntegerForKey(key)
	if score > oldScore then
		cc.UserDefault:getInstance():setIntegerForKey(key, score)
	end
end

function UserRecord:readIntegerFromUserDefault(key)
	return cc.UserDefault:getInstance():getIntegerForKey(key)
end

return UserRecord
